cdap-app-fabric/src/main/java/co/cask/cdap/metadata/profile/ProfileMetadataMessageProcessor.java

```diff
@@ -78,18 +83,19 @@
   private static final Gson GSON = ApplicationSpecificationAdapter.addTypeAdapters(
     new GsonBuilder().registerTypeAdapter(EntityId.class, new EntityIdTypeAdapter())).create();
 
-  private final MetadataStore metadataStore;
+  private final MetadataStorage metadataStorage;
   private final NamespaceTable namespaceTable;
   private final AppMetadataStore appMetadataStore;
   private final ProgramScheduleStoreDataset scheduleDataset;
   private final PreferencesTable preferencesTable;
 
-  public ProfileMetadataMessageProcessor(MetadataStore metadataStore, StructuredTableContext structuredTableContext) {
+  public ProfileMetadataMessageProcessor(MetadataStorage metadataStorage,
```

Comment:
Don't the metadata updates that happen here need to be published for audit?

Reply:
+1

Reply:
actually they are. Because the MetadataStorage here is not injected, but passed in through the constructor in MetadataMessageProcessor, which uses the Auditing one.